### ISSUE TITLE
Fix and improve mips-specific cctests

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#91]
We do have 1 bug on cctest test-assembler-mips.cc:

```
 cctest/test-assembler-mips64/MIPS14 ===
#
# Fatal error in ../test/cctest/test-assembler-mips64.cc, line 1281
# CHECK_EQ(static_cast<int32_t>(kFPUInvalidResult), t.round_invalid_result) failed
#   Expected: -1
#   Found: 2147483647
```

We also do not have very good coverage in that routine, we should improve it.

And in addition, we have several cctest files that are not implemented at all for mips64: `test-disasm-mips64.cc`, `test-macro-assembler-mips64.cc`, `test-code-stubs-mips.cc`

We should implement these, similar to *-mips.cc, but obviously on mips64 testing the 64-bit parts that we would not do in the mips files.

**Comment by xwafish** [comment]
Just add test-disasm-mips64.cc, test-macro-assembler-mips64.cc, but it does not  cover all, for example disasm does include float point instructions(so does mipsel). 
and need some more time to implemented  test-code-stubs, it is not implemented in mipsel too.

**Comment by paul99** [comment]
test-code-stubs-mips.cc was implemented Wed Jan 22 2014, with commit 2e6f966. It should be in the merged part of the tree. There were other changes to it later, in April that we will also have to follow.

And our coverage is certainly less than 100% in the others that you mention. We should improve these over time, but we need to start with something reasonable, for all.

**Comment by paul99** [comment]
there are also a lot of warnings coming from cctest/test-assembler-mips64.cc. Can you fix those?

Note: this is lower priority than bug-fixing on the mips64-gbl branch

**Comment by paul99** [comment]
Bojan - thought I'd assign this one to you, partly based on our discussion in https://github.com/paul99/v8m-rb/commit/d17ecdea4fb42447d05f8881fe7ee9e27fc26780

**Comment by BojanKV** [comment]
I've committed fixes for this issue in 306b3099c10f3390da0efc44c88b965c7e129b6a, and it is tested in simulator and on HW.. 
ADDU, ADDIU, SUBU and LUI are now implemented correctly in simulator to sign-extend result to 64bits.. 
There are no tests for 64bit instructions (like dclz, dext, ...), so additional tests should be implemented accordingly..

**Comment by paul99** [comment]
Completed. Closing.
